File: sdks/php/src/Webhooks/Requests/WebhooksControllerStripeRequest.php
<?php

namespace NovaBilling\Webhooks\Requests;

use NovaBilling\Core\Json\JsonSerializableType;

class WebhooksControllerStripeRequest extends JsonSerializableType
{
    /**
     * @var string $stripeSignature Stripe webhook signature
     */
    public string $stripeSignature;

    /**
     * @param array{
     *   stripeSignature: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->stripeSignature = $values['stripeSignature'];
    }
}

File: sdks/csharp/src/NovaBilling/Webhooks/Requests/WebhooksControllerStripeRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record WebhooksControllerStripeRequest
{
    /// <summary>
    /// Stripe webhook signature
    /// </summary>
    [JsonIgnore]
    public required string StripeSignature { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
